### PR TITLE
Rent model + migration

### PR DESCRIPTION
**Pull request by ffasitella**
### Summary

- Rent model
- Migration + foreign keys

### Trello card

https://trello.com/c/e7L5Yjyb/19-crear-modelo-de-rent

### Screenshots
Creating a new rent from the rails console:

image

### CODE REVIEW

File: app/models/rent.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Rent < ApplicationRecord
+  belongs_to :user
+  belongs_to :book
```
Comment (marcegl91) [inline app/models/rent.rb]: This is correct, but I can't ask a Book nor a User which rents is associated with, ie I can do `rent.user` and `rent.book` but I can't do `book.rents` nor `user.rents`.

File: db/migrate/20191120130458_create_rents.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateRents < ActiveRecord::Migration[5.1]
+  def change
+    create_table :rents do |t|
+      t.references :user, foreign_key: true, null: false
+      t.references :book, foreign_key: true, null: false
+      t.date :start_date, null: false
+      t.date :end_date, null: false
```
Comment (marcegl91) [inline db/migrate/20191120130458_create_rents.rb]: There is missing the timestamps that Rails need to keep track when a model is created or updated (ie the columns created_at and updated_at in the rents table).

File: app/models/book.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
 class Book < ApplicationRecord
+  has_one :rent, dependent: :destroy
```
Comment (marcegl91) [inline app/models/book.rb]: So the book can only be rented once, and after that no one can rent that book again.

File: app/models/user.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :registerable, :rememberable, :validatable
   include DeviseTokenAuth::Concerns::User
+  has_one :rent, dependent: :destroy
```
Comment (marcegl91) [inline app/models/user.rb]: Same as before, the user can only have one rent, after that it cannot rent anything else at all.

File: app/models/book.rb (hunk 3 above)
Comment (edkajimenez) [reply app/models/book.rb]: try with this https://guides.rubyonrails.org/association_basics.html having in mind that the same book can be rented by a lot of people through the time, also a person can rent many books

File: db/schema.rb [hunk 5]
```diff
@@ -42,9 +55,15 @@
     t.datetime "confirmed_at"
     t.datetime "confirmation_sent_at"
     t.json "tokens"
+    t.bigint "rent_id"
```
Comment (marcegl91) [inline db/schema.rb]: Do you need this column?

File: db/migrate/20191127131855_add_rent_to_user.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddRentToUser < ActiveRecord::Migration[5.1]
+  def change
+    add_reference :users, :rent, foreign_key: true
```
Comment (marcegl91) [inline db/migrate/20191127131855_add_rent_to_user.rb]: Do you need this migration?

File: db/migrate/20191127131936_add_rent_to_book.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class AddRentToBook < ActiveRecord::Migration[5.1]
+  def change
+    add_reference :books, :rent, foreign_key: true
```
Comment (marcegl91) [inline db/migrate/20191127131936_add_rent_to_book.rb]: Do you need this migration?

File: db/schema.rb [hunk 8]
```diff
@@ -24,6 +24,19 @@
     t.string "year", null: false
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.bigint "rent_id"
```
Comment (marcegl91) [inline db/schema.rb]: Do you need this column?

File: db/schema.rb [hunk 9]
```diff
@@ -34,8 +45,8 @@
     t.datetime "remember_created_at"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
-    t.string "first_name", null: false
-    t.string "last_name", null: false
+    t.string "first_name"
```
Comment (marcegl91) [inline db/schema.rb]: Why did you remove the null constrain from this fields? This was required in the card of the User creation.
